docs:Add instructions for adding support for new devices (#12)
BredOS/images: README.md

These are the officially supported images.</br>
For the downstream / experimental images, check the [downstream](https://github.com/BredOS/images/tree/downstream) branch.</br>

## Modifying the images to add new device

> We may want to add support for a new device or add support for a temperature or touch screen sensor on the i2c bus. This is different from the x86 world where the kernel is able to detect the hardware automatically with ACPI, etc. In the ARM world, the kernel needs to be told explicitly about the hardware. 

Reviewer: There is no need to avoid explaining the device tree, we can use it to refer to all the changes as alterations of it.
x86 modifications also require device tree overrides, it'd be best not to use it as a comparison point.
```suggestion
To define hardware onto the kernel (like for example to add a temperature sensor) the kernel on ARM systems needs to be explicitly told about the hardware using the kernel device tree.
```

> Currently, there are 2 ways to support a new device:

Reviewer: Referring to hardware modules as devices can confuse the reader, making him think of whole boards.
```suggestion
Currently, there are 2 ways to alter the device tree of a system:
```

 - Using the device tree overlay (DTO) method.
 - Recompiling the kernel

> The DTO method is the recommended way since it is more flexible and does not require recompiling the kernel which is a time consuming process. 

Reviewer: Just some rewording that I think feels more natural.
```suggestion
The DTO method highly recommended since it is much more flexible and does not require recompiling the kernel, which is very time consuming.
```

### Device Tree Overlay (DTO) method

> The DTO method is the recommended way to add support for a new device. The DTO method is a way to describe the hardware to the kernel. Depending on if your device is UEFI or U-Boot enabled, the steps to enable the DTO will be different.

Reviewer: Feels repetitive, not adding anything of worth really.
```suggestion
Depending on if your device is UEFI or U-Boot enabled, the steps to enable the DTO will be different.
```

Below are detailed instructions and examples for the DTO method:
 - https://wiki.bredos.org/en/how-to/how-to-setup-panthor
 - https://wiki.indiedroid.us/Nova/device-tree-overlay

### Recompiling the kernel

> The kernel recompilation method is the traditional way to add support for a new device.

Reviewer: No need to overcomplicate.
```suggestion
The kernel recompilation method is the traditional way to perform changes at the device tree.
```

Steps:
1. Clone the kernel source code from the [BredOS kernel repository](https://github.com/BredOS/linux-rockchip) and make the necessary changes.

2. Build the kernel with makepkg:

>     - Fork [sbc-pkgbuilds](https://github.com/BredOS/sbc-pkgbuilds/tree/main) and go to linux-rockchip-rkr3.

Reviewer: ```suggestion
    - Fork the `linux-rockchip-rkr3` branch of [sbc-pkgbuilds](https://github.com/BredOS/sbc-pkgbuilds/tree/main).
```

    - Run `makepkg -sr` to pull the kernel source code from the [BredOS kernel repository](https://github.com/BredOS/linux-rockchip), which is saved in `linux-rockchip-rkr3/src/linux-rockchip`.
    - Make the necessary changes in `linux-rockchip-rkr3/src/linux-rockchip`.
    - Note that your changes won't get saved if you remove the directory.
    - After making the changes, run `makepkg -srf` to make the kernel package in arch format.

3. Now you have the kernel package. Copy the kernel package to the temp repo /tmp/repo and update the repo database
```
mkdir /tmp/repo
cp *.pkg.tar.zst /tmp/repo
cd /tmp/repo
repo-add test-repo.db.tar.gz /tmp/repo/*pkg.tar.zst
```

4. Add the temp repo to the pacman.conf file under the specific board-cfg folder. For example, for orangepi 5, go to https://github.com/BredOS/images/tree/main/opi5-image and
then to <board-cfg>/pacman.conf.aarch64 where <board-cfg> is opi5-image:
```ini
[test-repo]
SigLevel = Never
Server = file:///tmp/repo
```

> Then just build image with mkimage.py in [mkimage](https://github.com/BredOS/mkimage)

Reviewer: ```suggestion
Then just build the image using [mkimage](https://github.com/BredOS/mkimage).
```
